dtests/src/test/java/io/snappydata/hydra/consistency/SnappyConsistencyTest.java

```diff
@@ -0,0 +1,405 @@
+
+  public Random rand = new Random();
+
+  public static void HydraTask_initialize() {
+    if (testInstance == null)
+      testInstance = new SnappyConsistencyTest();
+    if(derbyTestUtils == null)
+      derbyTestUtils = new DerbyTestUtils();
+  }
+
+  public static void HydraTask_performDMLOPsAndVerifyConsistency() {
+    testInstance.performOps(ConnType.JDBC);
+  }
+
+  public static void HydraTask_performDMLOPsAndVerifyConsistencyInJob() {
+    testInstance.performOps(ConnType.SNAPPY);
+  }
+
+  public static void HydraTask_performDMLOPsAndVerifyConsistencyInApp() {
+    testInstance.performOps(ConnType.SMARTCONNECTOR);
+  }
+
+  public void performOps(ConnType conn) {
+    int tid = getMyTid();
+    String dmlSql, selectSql;
+    String operation = SnappySchemaPrms.getDMLOperations();
+    String tableName;
+    int index;
+    switch (DMLOp.getOperation(operation)) {
+      case INSERT:
+        index = rand.nextInt(insertStmts.length);
+        dmlSql = insertStmts[index];
+        selectSql = SnappySchemaPrms.getAfterInsertSelectStmts()[index];
+        tableName = SnappySchemaPrms.getInsertTables()[index];
+        break;
+      case UPDATE:
+        index = rand.nextInt(updateStmts.length);
+        dmlSql = updateStmts[index];
+        selectSql = SnappySchemaPrms.getAfterUpdateSelectStmts()[index];
+        tableName = SnappySchemaPrms.getUpdateTables()[index];
+        break;
+      case DELETE:
+        index = rand.nextInt(deleteStmts.length);
+        dmlSql = deleteStmts[index];
+        selectSql = SnappySchemaPrms.getAfterDeleteSelectStmts()[index];
+        tableName = SnappySchemaPrms.getDeleteTables()[index];
+        break;
+      case PUTINTO:
+        index = rand.nextInt(putIntoStmts.length);
+        dmlSql = putIntoStmts[index];
+        selectSql = SnappySchemaPrms.getAfterPutIntoSelectStmts()[index];
+        tableName = SnappySchemaPrms.getInsertTables()[index];
+        break;
+      default:
+        Log.getLogWriter().info("Got invalid operation " + operation);
+        throw new TestException("Invalid operation type " + operation);
+    }
+    int batchSize = SnappySchemaPrms.getBatchSize();
+    index = Arrays.asList(SnappySchemaPrms.getTableNames()).indexOf(tableName);
+
+    if(operation.equalsIgnoreCase("delete") && dmlSql.contains("$counter")){
+      int delCounter = getDeleteCounter(index,batchSize);
+      dmlSql = dmlSql.replace("$counter",delCounter+ "");
+      selectSql = selectSql.replace("$counter", delCounter + "");
+    }
+
+    if (SnappySchemaPrms.isTestUniqueKeys() ) {
+      if (!(operation.equalsIgnoreCase("insert") || operation.equalsIgnoreCase("putinto")))
+        dmlSql = addTidToQuery(dmlSql, tid);
+      selectSql = addTidToQuery(selectSql, tid);
+    }
+    if (conn.equals(ConnType.JDBC)) {
+      ExecutorService pool = Executors.newFixedThreadPool(2);
+        Future<?> future1 =  pool.submit(new DMLOpsThread(tid, operation, batchSize, dmlSql,
```

Comment:
Can submit more insert/update tasks per threads or have more threads doing these operations in parallel